### ISSUE TITLE
[Repo Doctor] dx: Race Condition in Spinner State

### ISSUE THREAD

**Issue by glaucia86** [glaucia86/repocheckai#14]
## Summary
**Issue:** If `startSpinner()` returns falsy unexpectedly, subsequent calls fail silently.

## Evidence
```typescript
// src/ui/display/spinner.ts (inferred from usage pattern)
// Current pattern in agent.ts:
let spinner = !isSilent && !isJson ? startSpinner("Initializing...") : null;
// Later:
if (spinner) {
  updateSpinner("Creating session...");
}
```

## Impact
See analysis report for impact details.

## Recommended Fix
**Fix:**

---
_Generated by Repo Doctor._
